charts/wasp-routing-service-v2/templates/deployment.yaml

```diff
@@ -0,0 +1,184 @@
+      topologySpreadConstraints: {{- include "common.tplvalues.render" (dict "value" .Values.topologySpreadConstraints "context" .) | nindent 8 }}
+      {{- end }}
+      {{- if .Values.podSecurityContext.enabled }}
+      securityContext: {{- omit .Values.podSecurityContext "enabled" | toYaml | nindent 8 }}
+      {{- end }}
+      {{- if .Values.terminationGracePeriodSeconds }}
+      terminationGracePeriodSeconds: {{ .Values.terminationGracePeriodSeconds }}
+      {{- end }}
+      {{- if or .Values.initContainers .Values.initRawPayloads.enable }}
+      initContainers:
+        {{- if .Values.initRawPayloads.enable}}
+        - name: assert-raw-payloads
+          image: {{ template "wasp-routing-service.initRawPayloads.image" . }}
+          imagePullPolicy: {{ .Values.initRawPayloads.image.pullPolicy }}
+          env:
+            - name: KAFKA_BROKER
+              value: {{ include "wasp-routing-service.kafka.bootstrapBroker" . }}
+            - name: TOPIC
+              value: {{ .Values.initRawPayloads.topic }}
+            - name: PARTITIONS
+              value: {{ .Values.initRawPayloads.partitions | quote }}
+            - name: REPLICATION_FACTOR
+              value: {{ .Values.initRawPayloads.replicationFactor | quote }}
+          command: ['/bin/sh', '-c']
+          args: ['/opt/bitnami/kafka/bin/kafka-topics.sh --create --topic $(TOPIC) --bootstrap-server=$(KAFKA_BROKER) --partitions $(PARTITIONS) --replication-factor $(REPLICATION_FACTOR) --if-not-exists']
+        {{- end }}
+        {{- if .Values.initContainers }}
+          {{- include "common.tplvalues.render" (dict "value" .Values.initContainers "context" $) | nindent 8 }}
+        {{- end }}
+      {{- end }}
+      containers:
+        - name: wasp-routing-service
+          image: {{ template "wasp-routing-service.image" . }}
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          {{- if .Values.containerSecurityContext.enabled }}
+          securityContext: {{- omit .Values.containerSecurityContext "enabled" | toYaml | nindent 12 }}
+          {{- end }}
+          {{- if .Values.diagnosticMode.enabled }}
+          command: {{- include "common.tplvalues.render" (dict "value" .Values.diagnosticMode.command "context" $) | nindent 12 }}
+          {{- else if .Values.command }}
+          command: {{- include "common.tplvalues.render" (dict "value" .Values.command "context" $) | nindent 12 }}
+          {{- end }}
+          {{- if .Values.diagnosticMode.enabled }}
+          args: {{- include "common.tplvalues.render" (dict "value" .Values.diagnosticMode.args "context" $) | nindent 12 }}
+          {{- else if .Values.args }}
+          args: {{- include "common.tplvalues.render" (dict "value" .Values.args "context" $) | nindent 12 }}
+          {{- end }}
+          env:
+            - name: WASP_DEBUG
+              value: {{ ternary "true" "false" (or .Values.image.debug .Values.diagnosticMode.enabled) | quote }}
```

Comment:
Does the application respect this ENVAR?

Reply:
No it's there so when you inspect the container it's in debug mode

Reply:
Ah ok so its just a way of signalling to a user that this container is in debug mode.